ballerina/swiftmt_parser.bal

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import swiftmt.com.prowidesoftware.swift.io as prowide;
+
+import ballerina/data.xmldata;
+
+# Converts a SWIFT FIN message to the corresponding SWIFT MT message type or throws an error if the type is invalid.
+#
+# + finMessage - The input SWIFT FIN message string to be parsed and converted into a SWIFT MT message type.
+# + return - On success, returns one of the SWIFT MT message types (MT1XX, MT2XX, MT9XX, MTnXX).
+# In case of an error or unsupported message type, returns an error indicating the issue.
+public isolated function swiftMtParser(string finMessage) returns record {|anydata...;|}|error {
+    map<typedesc<record {}>> messageMapper = {"101": MT101Message, "102": MT102Message, "102STP": MT102STPMessage, "103": MT103Message, "103STP": MT103STPMessage, "103REMIT": MT103REMITMessage, "104": MT104Message, "107": MT107Message, "200": MT200Message, "201": MT201Message, "202": MT202Message, "202COV": MT202COVMessage, "203": MT203Message, "204": MT204Message, "205": MT205Message, "205COV": MT205COVMessage, "210": MT210Message, "900": MT900Message, "910": MT910Message, "920": MT920Message, "940": MT940Message, "941": MT941Message, "942": MT942Message, "950": MT950Message, "970": MT970Message, "971": MT971Message, "972": MT972Message, "973": MT973Message, "92": MTn92Message, "95": MTn95Message, "96": MTn96Message};
```

Comment:
Could make this a final and immutable map, at module-level?

Reply:
fixed in `99b8222`